staticaddr/address/manager.go

```diff
@@ -147,6 +171,7 @@ func (m *Manager) NewAddress(ctx context.Context) (*btcutil.AddressTaproot,
 		ProtocolVersion: version.AddressProtocolVersion(
 			protocolVersion,
 		),
+		InitiationHeight: m.currentHeight,
```

Comment:
Can `currentHeight` be uninitialized (0) at this point? IMHO we should either return an error here if currentHeight is 0 (if we are sure that this won't happen) or wait for its initialization (e.g. using a cond var).